ui/components/multichain/asset-picker-amount/swappable-currency-input/swappable-currency-input.tsx

```diff
@@ -93,6 +98,7 @@ export function SwappableCurrencyInput({
       )}
       asset={asset?.details}
       isSkeleton={isAmountLoading}
+      isMatchingUpstream={isSetToMax}
```

Comment:
Why not just call `const isSetToMax = useSelector(getSendMaxModeState);` in CurrencyInput?

Reply:
`CurrencyInput` should be decoupled from Send logic (see: https://github.com/MetaMask/metamask-extension/pull/23192#discussion_r1526686978)

Reply:
Gotcha, I think the naming could be improved then - maybe `isValueModifiedUpstream`

Reply:
Isn't starting with "is" convention?

Reply:
hmm good point - looks like is/has/can is the convention. I'm good with any prefix as long as the name is more descriptive

Reply:
Hmm yeah, it's tricky because it's technically always syncing with the upstream; this explicitly says that it will always *match* the upstream